cooked-validators/src/Cooked/Attack/TamperDatum.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cooked.Attack.TamperDatum where
+
+import Cooked.Attack.Common
+import Cooked.Tx.Constraints.Optics
+import Cooked.Tx.Constraints.Type
+import qualified Ledger.Typed.Scripts as Pl
+import Optics.Core
+import Type.Reflection
+
+-- | An attack that tries to change the datum on 'PaysScript' constraints with a
+-- prescribed tampering function, that only applies to datums of a certain type.
+tamperDatumAttack ::
+  forall a.
+  (Typeable a, PaysScriptConstrs a) =>
```

Comment:
Hmm, isn't `Typeable` a part of `PaysScriptConstrs`?

Reply:
You're right, it is! Good catch!

https://tweag.github.io/plutus-libs/Cooked-Tx-Constraints-Type.html#t:PaysScriptConstrs

Reply:
I'll take the liberty of changing that since Carl is probably already on holidays.